src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs

```diff
@@ -346,7 +354,7 @@ public override void Initialize([AllowNull] ViewWithData data)
         public Task<string> ExtractFile(IPullRequestFileNode file)
         {
             var path = Path.Combine(file.DirectoryPath, file.FileName);
-            return pullRequestsService.ExtractFile(repository, model.Head.Sha, path).ToTask();
+            return pullRequestsService.ExtractFile(repository, modelService, model.Head.Sha, path, file.Sha).ToTask();
         }
```

Comment:
You're totally right, we should be having methods that return `Task` when there's only one result, they are way better suited to convey the meaning of what's going on. If you want to change the signature of the `PullRequestService.ExtractFile` to return Task and use that everywhere, 👍